### PR TITLE
fix(curriculum): few small changes in test 7 and 8

### LINKED ISSUE #53153: Update descriptions, hint text and tests for Step 7 and 8 of  Todo App
### Describe the Issue

In the description, it is explained how `showModal()` method to be used on `confirmCloseDialog` and it is a built in method for `dialog` elements.

While in the hint it is being mentioned that  `showModal()` to be set on `confirmCloseDialog`, which I do not think is a correct statement. Setting `showModal()` is giving the illusion that `showModal() `to be assigned to `confirmCloseDialog` as value.

More appropriate hint would have been: 
> Your event listener should use arrow syntax to call the `showModal()` method on `confirmCloseDialog`.

`set` to be replaced with `call`

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-localstorage-by-building-a-todo-app/step-7

### Your code

```



```

### Expected behavior

Adding a more appropriate hint like: 
> Your event listener should use arrow syntax to call the `showModal()` method on `confirmCloseDialog`.

`set` to be replaced with `call`

### Screenshots

Screenshot 2024-01-12 at 9 14 16 PM

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by anikaitprasar**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #53153

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64e4eaaa9070a66aecbfe603.md [hunk 1]
```diff
@@ -27,7 +29,7 @@ Your event listener should listen for a `click` event.
 assert.match(code, /cancelBtn\.addEventListener\(('|"|`)click\1/)
 ```
 
-Your event listener should use arrow syntax to set the `close()` method on `confirmCloseDialog`. Don't use curly braces.
+Your event listener's callback function should call the `close()` method on `confirmCloseDialog` element.
 
 ```js
 assert.match(code, /cancelBtn\.addEventListener\(('|"|`)click\1\,\s*\(\)\s*\s*=>\s*confirmCloseDialog\.close\(\)\s*\);?/)
```
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64e4eaaa9070a66aecbfe603.md:35]:
```suggestion
assert.match(code, /cancelBtn\.addEventListener\(('|"|`)click\1\s*\,\s*\(\s*\)\s*=>\s*(\{\s*confirmCloseDialog\.close\(\s*\)\s*;?\s*\}|confirmCloseDialog\.close\(\s*\))\s*\)\s*;?/)
```
Here too.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64e4e7bbedb22d6939001ad3.md [hunk 2]
```diff
@@ -7,9 +7,13 @@ dashedName: step-7
 
 # --description--
 
-Add a `click` event listener to the `closeTaskFormBtn` variable, then use the `showModal()` method on your `confirmCloseDialog` variable. This will display a modal with the `Discard` and `Cancel` buttons.
+The HTML `dialog` element has a `showModal()` method that can be used to display a modal dialog box on a web page.
 
-`showModal()` is a method associated with the HTML `dialog` element. It is used to display a modal dialog box on a web page.
+`dialogElement.showModal();`
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64e4e7bbedb22d6939001ad3.md:12]: This should be inside a code fence like this 

````
```js
dialogElement.showModal();
```
````

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64e4eaaa9070a66aecbfe603.md [hunk 3]
```diff
@@ -7,11 +7,13 @@ dashedName: step-8
 
 # --description--
 
-If the user clicks the `Cancel` button, you want to cancel the process (close the modal showing the two buttons) so the user can continue editing.
+If the user clicks the `Cancel` button, you want to cancel the process and close the modal so the user can continue editing. The HTML `dialog` element has a `close()` method that can be used to close a modal dialog box on a web page.
 
-Add a `click` event listener to the `cancelBtn` element, then use the `close()` method on the `confirmCloseDialog` variable.
+`dialogElement.close();`
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64e4eaaa9070a66aecbfe603.md:12]: same issue 

This should be in a code fence like this 

````
```js
dialogElement.close();
```
````

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64e4e7bbedb22d6939001ad3.md (hunk 2 above)
Comment (anikait-prasar) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64e4e7bbedb22d6939001ad3.md]: Done. Thanks and I apologize for the inconvenience caused.

**Review by jdwilkin4 — APPROVED** [review]
Looks good 👍

**Review by Sembauke — APPROVED** [review]

**Pull request merged**